vm/isolated/isolated.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package isolated
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/google/syzkaller/pkg/config"
+	. "github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/vm/vmimpl"
+)
+
+func init() {
+	vmimpl.Register("isolated", ctor)
+}
+
+type Config struct {
+	Target_Addr string // target machine
+	Target_Dir  string // directory to copy/run on target
+}
+
+type Pool struct {
+	env *vmimpl.Env
+	cfg *Config
+}
+
+type instance struct {
+	cfg    *Config
+	closed chan bool
+	debug  bool
+	sshkey string
+	port   int
+}
+
+func ctor(env *vmimpl.Env) (vmimpl.Pool, error) {
+	cfg := &Config{}
+	if err := config.LoadData(env.Config, cfg); err != nil {
+		return nil, err
+	}
+	if cfg.Target_Addr == "" {
+		return nil, fmt.Errorf("config param targethost is empty")
+	}
+	if cfg.Target_Dir == "" {
+		return nil, fmt.Errorf("config param targetdir is empty")
+	}
+	// sshkey is optional
+	if env.Sshkey != "" && !osutil.IsExist(env.Sshkey) {
+		return nil, fmt.Errorf("ssh key '%v' does not exist", env.Sshkey)
+	}
+	env.Target_WorkDir = cfg.Target_Dir
```

Comment:
Instead of threading workdir through multiple components back to the syz-fuzzer we start in Run, I think it's better to do what adb does in Run:
```
exec.Command(inst.adbBin, "-s", inst.device, "shell", "cd /data; "+command)
```

We've tried hard to keep VM-independent parts as VM-independent as possible (it was a mess before). Adding something to the common VM environment that is needed only for 1 VM type is kind of a red flag.

We could also chdir into the current binary dir in syz-fuzzer, which would probably be reasonable (not sure if won't break something). But there are also other binaries (not necessary syz-fuzzer) and trying to give them -workdir flag or expecting them to chdir is pointless. `shell cd` will handle other binaries as well.